test/integration/component/test_template_tags.py

```diff
@@ -0,0 +1,437 @@
+        )
+
+        self.verify_upload_iso(self.iso)
+        list_iso_response = self.list_isos(None)
+        self.assertEqual(
+            isinstance(list_iso_response, list),
+            True,
+            "No iso found"
+        )
+
+        # Step 4
+        self.update_iso(self.iso.id, iso_tag)
+
+        # Step 5
+        list_iso_response = self.list_isos(iso_tag)
+
+        # Step 6
+        self.assertEqual(
+            isinstance(list_iso_response, list),
+            True,
+            "Iso tag is not updated properly to %s" % iso_tag
+        )
+        self.assertEqual(
+            len(list_iso_response),
+            1,
+            "Check template available in List ISOs"
+        )
+
+        # Step 7
+        list_template_response = self.list_templates("dummy")
+        self.assertEqual(
+            list_template_response,
+            None,
+            "Template tag should not be updated to dummy"
+        )
+
+        list_iso_response = self.list_isos("dummy")
+
+        # Step 8
+        self.assertEqual(
+            list_iso_response,
+            None,
+            "ISO tag should not be updated to dummy"
+        )
+
+    @attr(tags=["advanced", "basic", "sg"], required_hardware="false")
+    def test_02_register_template_with_tags(self):
+        """
+        Register template and iso with new template tags
+        Verify that it can be registered with tags
+        Listing the templates/iso with tags should return the result
+
+        Steps:
+        1. Register the template with tag "REGISTER_TEMPLATE"
+        2. Register the iso tag of uploaded iso as "REGISTER_ISO"
+        3. List all the templates matching the tag "REGISTER_TEMPLATE"
+        4. Make sure that the count is 1
+        5. List all the isos matching the tag "REGISTER_ISO"
+        6. Make sure that the count is 1
+        7. Update the template tag to "dummy"
+        8. List the templates matching tag "dummy"
+        9. Make sure that the count is 1
+        10. List the iso matching tag "dummy"
+        11. Make sure that the count is 1
+        12. List all the templates matching the tag "REGISTER_TEMPLATE"
+        13. Make sure that the count is 0
+        14. List all the isos matching the tag "REGISTER_ISO"
+        15. Make sure that the count is 0
+        :return:
+        """
+        template_tag = "REGISTER_TEMPLATE"
+        iso_tag = "REGISTER_ISO"
+
+        # Step 1
+        self.services["test_templates"]["kvm"]["templatetag"] = template_tag
+        template1 = Template.register(self.apiclient,
+                                      self.services["test_templates"]["kvm"],
+                                      zoneid=self.zone.id,
+                                      domainid=self.domain1.id,
+                                      account=self.account1.name,
+                                      hypervisor=self.hypervisor.lower()
+                                      )
```

Comment:
I see no instance level `setup()` and `tearDown()`, so this template will never be cleaned ip.